client/components/data/sync-active-theme/index.tsx

```diff
@@ -1,36 +1,54 @@
+import { translate } from 'i18n-calypso';
 import { useEffect, useState } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 import { waitFor } from 'calypso/my-sites/marketplace/util';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
 import { requestActiveTheme } from 'calypso/state/themes/actions';
 import { isRequestingActiveTheme, getActiveTheme } from 'calypso/state/themes/selectors';
 
 interface SyncActiveThemeProps {
 	siteId: number;
 	themeId: string;
-	onThemeActive: CallableFunction;
+	onAtomicThemeActive: CallableFunction;
+	onFailure?: CallableFunction;
 	maxAttempts?: number;
+	delay?: number;
 }
 
 const SyncActiveTheme = ( {
 	siteId,
 	themeId,
-	onThemeActive,
-	maxAttempts = 20,
+	onAtomicThemeActive,
+	onFailure,
+	maxAttempts = 50,
+	delay = 2,
 }: SyncActiveThemeProps ) => {
 	const dispatch = useDispatch();
 	const requestStarted = useSelector( ( state ) => isRequestingActiveTheme( state, siteId ) );
 
 	const activeTheme = useSelector( ( state ) => getActiveTheme( state, siteId ) );
 
 	const [ attempts, setAttempts ] = useState( 0 );
+	const [ exceededMaxAttempts, setExceededMaxAttempts ] = useState( false );
+
+	useEffect( () => {
+		if ( exceededMaxAttempts ) {
+			dispatch( recordTracksEvent( 'calypso_sync_active_theme_exceeded_max_attempts' ) );
```

Comment:
Can we add some props to the event so it's more useful?
```suggestion
			dispatch( recordTracksEvent(
				'calypso_sync_active_theme_exceeded_max_attempts',
				{
					delay,
					max_attempts,
					theme: themeId,
				}
			) );
```
We'll also need to add these to the deps for the effect. I did consider adding `attempts`, but that would trigger the effect every time `attempts` is incremented, which we don't want.